src/Embeddings/VectorStores/Elasticsearch/ElasticsearchVectorStore.php

```diff
@@ -0,0 +1,155 @@
+                            ],
+                            'content' => [
+                                'type' => 'text',
+                            ],
+                            'formattedContent' => [
+                                'type' => 'text',
+                            ],
+                            'sourceType' => [
+                                'type' => 'keyword',
+                            ],
+                            'sourceName' => [
+                                'type' => 'keyword',
+                            ],
+                            'hash' => [
+                                'type' => 'keyword',
+                            ],
+                            'chunkNumber' => [
+                                'type' => 'integer',
+                            ],
+                        ],
+                    ],
+                ],
+            ]);
+        }
+    }
+
+    public function addDocument(Document $document): void
+    {
+        $this->client->index([
+            'index' => $this->indexName,
+            'body' => [
+                'embedding' => $document->embedding,
+                'content' => $document->content,
+                'formattedContent' => $document->formattedContent ?? '',
+                'sourceType' => $document->sourceType,
+                'sourceName' => $document->sourceName,
+                'hash' => $document->hash,
+                'chunkNumber' => $document->chunkNumber,
+            ],
+        ]);
+    }
+
+    /**
+     * @param  Document[]  $documents
+     */
+    public function addDocuments(array $documents, int $numberOfDocumentsPerRequest = 0): void
+    {
+        $params = ['body' => []];
+        foreach ($documents as $document) {
+            $params['body'][] = [
+                'index' => [
+                    '_index' => $this->indexName,
+                ],
+            ];
+            $params['body'][] = [
+                'embedding' => $document->embedding,
+                'content' => $document->content,
+                'formattedContent' => $document->formattedContent ?? '',
+                'sourceType' => $document->sourceType,
+                'sourceName' => $document->sourceName,
+                'hash' => $document->hash,
+                'chunkNumber' => $document->chunkNumber,
+            ];
+
+        }
+        $this->client->bulk($params);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param  array{filter?: string|array<string, mixed>}  $additionalArguments
+     */
+    public function similaritySearch(array $embedding, int $k = 4, array $additionalArguments = []): array
+    {
+        $searchParams = [
+            'index' => $this->indexName,
+            'body' => [
+                'knn' => [
+                    'field' => 'embedding',
+                    'query_vector' => $embedding,
+                    'k' => $k,
+                    'num_candidates' => 50,
```

Comment:
what is this 50?

Reply:
It's an important parameter to tune approximate kNN for speed or accuracy. I forgot to make it a parameter ! 👍 
![Capture d’écran 2023-10-10 à 11 12 46](https://github.com/theodo-group/LLPhant/assets/55784802/2a8c2ca7-f3fc-4011-99ab-446148fb6155)

Reply:
Ok maybe 50 is too much. We could make it relative to $k like 4*$k ?
I don't know what it is suggested in the doc?

Reply:
The doc use either 50 or 100. I don't think making it proportional to the number of returned result make sens, i added the possibility to define it in the $additionalArguments with a default value of 50.

Reply:
If the user need a lot of result, he needs accuracy for all the result : num_candidates = $k * 4
But we should have a minimum accuracy for low $k number ➔ num_candidates = 50
So : **$num_candidates = max(50, $k * 4)**